setup.py

```diff
@@ -26,7 +26,7 @@
           'joblib==1.1.0',
           'PyMySQL==0.7.11',
           'psycopg2-binary==2.8.6',
-          'snowflake-connector-python[pandas]==2.4.6',
+          'snowflake-connector-python[pandas]==3.0.2',
```

Comment:
@florian-ernst-alan I upgraded snowflake connector so that it would work on my mac. We could skip this though if we are concerned it could break things

Reply:
@pvonglehn I would revert this before merging, since it's touching the internal packages of pipelinewise. I will update pipelinewise in the coming weeks in any case.